packages/nocodb/src/db/conditionV2.ts

```diff
@@ -740,7 +759,9 @@ const parseConditionV2 = async (
                   // else
                   qb = qb.where(knex.raw('??::date = ?', [field, val]));
                 } else {
-                  qb = qb.where(knex.raw('DATE(??) = DATE(?)', [field, val]));
+                  qb = qb.where(
+                    knex.raw("DATE(??, 'localtime') = DATE(?)", [field, val]),
```

Comment:
Does all databases(sqlite, mysql and mssql) take `localtime` as second argument ?

Reply:
MySQL was excluded from the beginning. Here, I am only testing SQLite and have not considered MSSQL. I can modify this through conditional statements.

Reply:
his only fixes the issue. I feel this part can be abstracted and optimized further.